### DISCUSSION TITLE
facing error while calling bignum prime generation method

### DISCUSSION THREAD

**Discussion by ranjithravitk** [openssl/openssl#24707] [Q&A] [answered]
When I tried to call the BN_generate_prime_ex2(prime, 2048, 1, t1, t2, NULL, ctx) from a custom provider. 
I am facing :

80FBE4F7FF7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:355:Global default library context, Algorithm (CTR-DRBG : 0), Properties (<null>)
80FBE4F7FF7F0000:error:12000090:random number generator:rand_new_drbg:unable to fetch drbg:crypto/rand/rand_lib.c:655:

Any help to identify the root cause will be useful. Thanks!!

**Comment by paulidale** [comment]
It is trying to locate either CTR-DRBG or one of the algorithms it requires (AES-256-CTR and AES-256-ECB) and failing.  Does your provider implement all three?  If not you need the FIPS+base or default providers loaded as well.

**Reply by ranjithravitk** [reply]
I am implementing elgamal encryption scheme as provider and trying to call this method to generate the prime number. I downloaded the openssl source code and compiled it with debug flags to understand the integration between openssl and my provider. While doing so, i was facing this issue. But now i tried it by installing the openssl and run the debugger in my provider which executed successfully. So i believe there is an issue with the source code i took or with my configuration. any suggestion on how i can setup the openssl source code as well to understand the integration between provider and openssl?

**Reply by paulidale** [reply, marked as answer]
You must not have told OpenSSL to load the default provider before you installed it.  Once installed, it's probably picking this up from the configuration file and your provider gets loaded in addition.

**Reply by ranjithravitk** [reply]
oh ok. Thank you!! I will check it out.
